examples/advanced_diffusion_training/train_dreambooth_lora_sdxl_advanced.py

```diff
@@ -1790,6 +1832,18 @@ def compute_text_embeddings(prompt, text_encoders, tokenizers):
         disable=not accelerator.is_local_main_process,
     )
 
+    def get_sigmas(timesteps, n_dim=4, dtype=torch.float32):
+        sigmas = noise_scheduler.sigmas.to(device=accelerator.device, dtype=dtype)
+        schedule_timesteps = noise_scheduler.timesteps.to(accelerator.device)
+        timesteps = timesteps.to(accelerator.device)
+
+        step_indices = [(schedule_timesteps == t).nonzero().item() for t in timesteps]
+
+        sigma = sigmas[step_indices].flatten()
+        while len(sigma.shape) < n_dim:
+            sigma = sigma.unsqueeze(-1)
+        return sigma
```

Comment:
Add a todo here to revisit different sampling algorithms for obtaining the sigmas. Ideally, Tero Karras should just drop a PR here just sayin' 🤷‍♂️